docs/adding_benchmarks_guide.md

```diff
@@ -0,0 +1,282 @@
+   Example:
+   ```go
+   grpcServer := grpc.NewServer(grpc.UnaryInterceptor(otelgrpc.UnaryServerInterceptor()))
+   ```
+3. If your function is a client, include client interceptor instrumentation when connecting to a 
+   server. Example:
+   ```go
+   conn, err := grpc.Dial(fmt.Sprintf("%v:%v", ps.consumerAddr, ps.consumerPort), 
+                   grpc.WithInsecure(), grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()))
+   ```
+
+You can check the 
+[producer in our serving example](/function-images/tests/chained-function-serving/producer/producer.go) 
+for example usage of this utility for both server and client behaviour.
+
+## Serving
+To compose functions with serving we make use of the 
+[Knative Serving component](https://knative.dev/docs/serving/). Each of our functions will 
+effectively be a server for the functions that come before it, and a client of the functions that 
+come after. For example in a simple chain of functions `A -> B -> C`, B would be a client of 
+function C and a server for function A.
+
+our example of function composition using serving can be found 
+[here](function-images/test/chained-function-serving), and additional CI implementation which shows
+ how this code is executed can be found 
+[here](https://github.com/ease-lab/vhive/blob/main/.github/workflows/function-composition-bench.yml).
+ This example implements a simple Client -> Producer -> Consumer function chain, whereby the client
+ triggers the producer function to generate a random string, and the consumer consumes said string
+ (by logging it to a file).
+
+To deploy your workload with serving you will need to:
+- Implement a remote procedure call system (e.g. `gRPC`)
+- Dockerize your functions
+- Write Knative manifests
+
+### RPC Implementation
+As mentioned in the general guidelines, your functions will communicate with RPC calls. In 
+[our serving example](function-images/test/chained-function-serving) we use gRPC with protobuf to 
+achieve this.
+
+In serving, each "link" in your chain of functions needs to implement a protobuf service. For 
+example in a chain `A -> B -> C` the A-B link will be one service and the B-C link will be a second
+ service. You should define a 
+ [proto file](https://developers.google.com/protocol-buffers/docs/proto3) for these services, and 
+ use it to generate code with `protoc`. You can see an example 
+ [here](function-images/test/chained-function-serving/proto).
+
+Implement the appropriate server or client service from the generated proto code in your functions 
+and remember that some functions such as the 
+[producer in our example](function-images/test/chained-function-serving/producer/producer.go) will 
+need to be both a server of one proto service and a client of another proto service simultaneously.
+
+### Dockerizing Functions
+Every function needs to be built as a docker image. Refer to the general guidelines.
+
```

Comment:
remove because it's a repetition, just add give a reference to the images of the example somewhere above.